### DISCUSSION TITLE
AsyncIOs in Old and New Viz Modules?

### DISCUSSION THREAD

**Discussion by Anonymous160609** [mesa/mesa#2152] [Q&A] [answered]
Hello MESA community. I am a part-time researcher on working on combining Space Syntax and ABM, for a publication then for my PhD application. 

My primary task is modelling the network movements of 75,000 residents (much like George Mason UNet), and optimally in a smooth matter. I suppose in my case multiprocessing and MESA-Frame have to be considered. However, it remains the question in viz efficiencies while Solara seems have no special optimizations in async data transmissions. 

In short do you fellows recommend me to stick onto the old viz in D3JS and Tornado, or to switch to the new fashion with Solara? I spent about one month in reading D3Js and Tornado to understand the MESA source code, yet, long term speaking I suppose the new paradigm in Matplotlib is more benefitial to future PhD studies, in relation to other analytic packages.

Kinds

**Comment by rht** [comment]
I recommend using Solara. The Tornado viz won't have new features from the new development in space such as `PropertyLayer` or Cells. It's a lot easier to build new features on the Solara version.

**Comment by Anonymous160609** [comment, marked as answer]
Solara provides .task and .use_task for async processings, while some its depedent webframes e.g. FASTapi also implicit these performances.
